Add Sentry Implementation to Project

tan-nguyen090102 opened the pull request.

tan-nguyen090102 committed:
Finish setup Sentry and test error reporting on temp account

tan-nguyen090102 committed:
Fix hydration mismatch on Navbar and public page

tan-nguyen090102 committed:
Add Error page for exception and change consistency in login.post

tan-nguyen090102 committed:
Add Sentry implementation to project

tan-nguyen090102 committed:
Add filter to only allow certain errors to be reported

tan-nguyen090102 committed:
Revise status code to send report

tan-nguyen090102 committed:
Merge branch 'filter-error-report' into add-sentry

tan-nguyen090102 committed:
Add transaction name to error package

tan-nguyen090102 committed:
Merge branch 'add-error-transaction-name' into add-sentry

tan-nguyen090102 committed:
Update the example dotenv file

tan-nguyen090102 committed:
Separate client and server Sentry report to different project DSN, update dotenv file

The sentry config file read both CLIENT and API DSN in dotenv file.

tan-nguyen090102 committed:
Add Sentry capture exception on expected catches on client setting.vue

tan-nguyen090102 committed:
Add breadcrump config to filter out the console log

This beforeBreadcrumb will get rid of the trace for console logs on Sentry.io dashboard.

tan-nguyen090102 committed:
Update the condition on status code checking to reduce the code list

tan-nguyen090102 committed:
Remove the commented throw exception in login.post

tan-nguyen090102 committed:
Remove event transaction name

Since the client and the api now point at different DSN project, no need for name to distinguish events apart.

TesfayeY committed:
Added routing and context to sentry events

It includes stable route names to avoid flooding with unique names and simplifies filtering by tags

TesfayeY committed:
Setting tags for path

TesfayeY committed:
Cleaning up cookie and setting email tag after successful login for events.

It cleans cookie info after logout for and upon successful login it tags all events sent to sentry associated with email tag.

sgdavis1 reviewed (changes requested):
Please address and adjust. Thank you and great work so far!

TesfayeY committed:
Improved Sentry user Context for robust error tracking and reporting

Ensures user email and ID are promptly associated with Sentry events.
Leverages server-set cookies for user context on subsequent page loads.

tan-nguyen090102 committed:
Update dotenv file documentation

Bring out Sentry organization, project, and environment setup outside

tan-nguyen090102 committed:
Add sourcemap documentation and details for client and server

tan-nguyen090102 committed:
Change environment to use the dotenv value for both Senry configs

tan-nguyen090102 committed:
Remove the library override for vercel and ofetch

Those library override are mainly for implementation for Next.js and not affect the Nuxt.js framework.

tan-nguyen090102 committed:
Bring the hardcoded environment name out to dotenv, update status description

Update the description for denied status code, though this may be changed on future refactor

tan-nguyen090102 committed:
Update condition on beforeSend and beforeBreadcrumb with ternary format

tan-nguyen090102 committed:
Add version tag to each Sentry reporting from package.json

tan-nguyen090102 committed:
Fix sentry.server.config.ts to have the package.json imported

tan-nguyen090102 committed:
Change the logo reference from remote URL to static image file

The image is located at src/public/img folder.

tan-nguyen090102 commented on src/layouts/public.vue:
```diff
@@ -31,6 +33,8 @@
   </template>
 
   <script setup lang="ts">
+  import { ref, computed } from 'vue';
```
Yes, Tesfaye and I noticed that there was a hydration mismatch when reloading the page (the sun icon on the dark mode instead of the moon). The original code to update isDark variable use computed function but not importing it here, and the issue was resolved when I add the import statement and wrap the icon rendering part to be ClientOnly.

TesfayeY committed:
Included route and username information

TesfayeY committed:
Give context to logout errors

Gives context to session issues, token problem and backend failure

tan-nguyen090102 committed:
Revert the error message changes to be similar with devel branch

tan-nguyen090102 committed:
Add option to enable and disable status code filtering.

Update dotenv file to have the environment variable.

tan-nguyen090102 committed:
Change the remote reference image to the static file image

TesfayeY committed:
Error Page context update

Included more context and tag information to errors originating from the error page

TesfayeY committed:
Missing closing bracket fix

Line 74 had a missing closing bracket, this will fix it.

TesfayeY committed:
Renamed and Removed overriding calls

Renamed route_path and removed params call as it doesnt provide any importannt info. Can be easily included if needed in future.

tan-nguyen090102 committed:
Update Sentry client config file to accept dotenv value now.

Using dotenv.example to set up all parameters to use for Sentry.

tan-nguyen090102 committed:
Remove the character hanging out of the function

tan-nguyen090102 committed:
Uncomment the tracesSampler function, add more documents on status code.

tan-nguyen090102 commented on sentry.client.config.ts:
```diff
@@ -0,0 +1,57 @@
+
+  // We recommend adjusting this value in production, or using tracesSampler
+  // for finer control
+  tracesSampleRate: 1.0,
```
At the moment I am able pull these value out to the dotenv file and has some mechanism to default to disable if leaving the variable blank, so those var can be omitted and freely adjusted. But I believe there is a better way to handle these without having many dotenv variables to add.

TesfayeY committed:
feature: Centalized SENTRY ERROR LOGGING

Introduces useErrorLogger composable, abstracting direct calls to Sentry.captureException(Error).

TesfayeY committed:
Introduces the 'useErrorLogger' composable abstract to Sentry error reporting.

tan-nguyen090102 committed:
Move server tracing and debug to use dotenv file

tan-nguyen090102 commented on dotenv.example:
```diff
@@ -25,6 +25,10 @@ APP_PORT=
 DB_PORT=
 
+# The public keys of the Sentry portal
+SENTRY_DSN_CLIENT=
```
Since our last discussion on Slack I split the Sentry project into two for client and server separately. But I have found this discussion with the Sentry developer on GitHub that may affect my past decision. https://github.com/getsentry/sentry-javascript/discussions/14499

tan-nguyen090102 committed:
Update README.md for Sentry implementation.

tan-nguyen090102 committed:
Remove the example value from dotenv.example file

sgdavis1 reviewed (changes requested):
A few more changes but this is headed in the right direction. Thanks guys!

tan-nguyen090102 commented on README.md:
```diff
@@ -148,7 +148,25 @@ Changes are handled via API requests, handlers located in server folder
+2. Create two new projects: one for client and one for server.
+3. Navigate to the project settings and copy the Client ID (DSN) for each project and paste to the local `.env` file.
+4. Also copy the project slug and navigate to organization setting to copy the organization slug and paste those to the local `.env` file.
+5. Make sure that the Sentry SDK is install. Copy this command if you haven't install one:
```
It just more like a redundant instruction if somehow the package is not installed correctly on `npm install`, but I can remove it for clarity.

tan-nguyen090102 committed:
Update README.md on Sentry to remove redundant instruction

tan-nguyen090102 committed:
Update dotenv file and adjust some naming

tan-nguyen090102 committed:
Change default sentry environment to be production if omitted.

tan-nguyen090102 commented on nuxt.config.ts:
```diff
@@ -2,18 +2,37 @@
+      sentry: {
+        dsnClient: process.env.SENTRY_DSN_CLIENT,
+        dsnApi: process.env.SENTRY_DSN_API,
+        environment: process.env.ENV,
```
Thank you for your advice, I will rename these. I just found that if there is no string supplied to sentry environment, it automatically defaults to production env on the dashboard. So I would like to make the production as default if not set.

tan-nguyen090102 committed:
Change tag version to release fields for Sentry reporting

The release is drawn from the version field in package.json

tan-nguyen090102 commented on sentry.client.config.ts:
```diff
@@ -0,0 +1,118 @@
+
+  try {
+    defaultTraceRate = parseFloat(useRuntimeConfig().public.sentry.traceConfig.defaultSamplingRate.toString());
+  } catch {
```
Thank you for your advice!

sgdavis1 reviewed (approved):
Great! Well done guys. Let's call this version 0.2.0 :)

sgdavis1 commented:
I meant 0.3.0, haha
